processor.go

```diff
@@ -215,6 +215,23 @@ func (proc *Processor) Encode(text string) []Token {
 		return -1
 	})
 
+	buf := make([]byte, 64) // reusable buffer for merging symbols
+	mergeSymbols := func(x, y symListElem) (string, int, bool) {
+		sz := len(x.symbol) + len(y.symbol)
+		if cap(buf) < sz {
+			buf = append(buf[:0], make([]byte, sz)...)
+		}
+		buf = buf[:cap(buf)]
```

Comment:
Why is this line needed?

Reply:
For the map lookup, we do:

```go
buf = buf[:sz]
```

but copy respects slice length, so this might make buf too small. So we reset buf to be as long as its cap before copying.